### DISCUSSION TITLE
express  import issues

### DISCUSSION THREAD

**Discussion by tyrads** [oven-sh/bun#6925] [General]
I am building an express api with bun, i am getting following error on the following import

`import { Request } from "express";
`

`SyntaxError: Import named 'Request' not found in module '/Users/abcd/ccc/test-project/node_modules/express/index.js'.
      at processTicksAndRejections (:61:76)
56 |         }
57 |       }
58 |       return new FixedQueue;
59 |     }();
60 | 
61 |     function processTicksAndRejections() {`

**Comment by aqibgatoo** [comment]
I am facing the same issue?

**Comment by Jarred-Sumner** [comment]
The error message is correct - it is not valid in Node.js either:
```js
❯ node req.mjs
file:///Users/jarred/Desktop/req.mjs:1
import { Request } from "express";
         ^^^^^^^
SyntaxError: Named export 'Request' not found. The requested module 'express' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'express';
const { Request } = pkg;
```

Is `Request` a type? If so, you could do

```js
import { type Request } from "express";
```

**Comment by aqibgatoo** [comment]
Thanks @Jarred-Sumner, importing as a type did work, but just curious why did it work in node, something i was doing in my node project, was it something that that ts-node took care of?
<img width="708" alt="image" src="https://github.com/oven-sh/bun/assets/4655062/bd356da3-1a48-4336-9dd7-804cc99185a8">
